### PR TITLE
Added the functionality to duplicate items and questions in a single api call

### LINKED ISSUE #313: Just make one API call to `/plios` for duplicating plios

### LINKED ISSUE #331: Create a separate instance of video upon duplicating

### LINKED ISSUE #332: Create a separate instance of image upon duplicating

### PR DESCRIPTION
**Pull request by gurleen3**
Fixes #313 
Fixes #331 
Fixes #332

## Summary

Earlier, 3 seperate api calls took place when a user clicked on duplicate option of the plio. Added the functionality to duplicate items and questions in a single api call to plios endpoint.

## Test Plan

- [x] Wrote tests
- [x] Tested locally
- [ ] Tested on staging
- [ ] Tested on production
- [ ] ~If adding or removing any environment variable:~
    - [ ] update `docs/ENV.md`
    - [ ] update the environment variables for staging and production
- [ ] ~If changes in DB, update [DB schema](https://docs.google.com/presentation/d/15c45HbHphh4w3guaFV3BgEA5nMKd-4J35JdUKdrpPzo/edit#slide=id.gc8c9ec0b5e_0_0) and BigQuery (staging and prod)~

### CODE REVIEW

File: plio/urls.py [hunk 1]
```diff
@@ -62,7 +62,6 @@
 api_router.register(r"organizations", OrganizationViewSet, basename="organizations")
 api_router.register(r"users", UserViewSet, basename="users")
 api_router.register(r"videos", VideoViewSet, basename="videos")
-# https://stackoverflow.com/questions/48548622/base-name-argument-not-specified-and-could-not-automatically-determine-the-name
```
Comment (dalmia) [inline plio/urls.py:65]: why have you deleted this?

File: plio/views.py [hunk 2]
```diff
@@ -697,59 +716,6 @@ def get_queryset(self):
...
-
-        items_to_delete = Item.objects.filter(pk__in=ids_to_delete)
-        if len(items_to_delete) != len(ids_to_delete):
-            return Response(
-                {"detail": "one or more of the ids provided do not exist"},
-                status=status.HTTP_404_NOT_FOUND,
-            )
-
-        items_to_delete.delete()
-        return Response("deletion successful")
```
Comment (dalmia) [inline plio/views.py:751]: why are you deleting this?

File: plio/views.py [hunk 3]
```diff
@@ -813,26 +748,3 @@ class ImageViewSet(viewsets.ModelViewSet):
 
     queryset = Image.objects.all()
     serializer_class = ImageSerializer
-    permission_classes = [IsAuthenticated]
```
Comment (dalmia) [inline plio/views.py:816]: why are you deleting this line?

File: plio/tests.py [hunk 4]
```diff
@@ -343,13 +343,70 @@ def test_user_can_play_own_private_plio(self):
 
     def test_duplicate(self):
         plio = Plio.objects.filter(created_by=self.user).first()
```
Comment (dalmia) [inline plio/tests.py:345]: we don't need this right?

File: plio/tests.py [hunk 5]
```diff
@@ -343,13 +343,70 @@ def test_user_can_play_own_private_plio(self):
...
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.data["item"], new_item_id)
+            self.assertEqual(response.data["type"], old_question.type)
+
+            if old_question.image is not None:
+                self.assertEqual(
+                    ImageSerializer(old_question.image).data["url"],
+                    response.data["image"]["url"],
+                )
+        response = self.client.post(f"/api/v1/plios/{plio.uuid}/duplicate/")
```
Comment (dalmia) [inline plio/tests.py:409]: what is the use of this line?

File: plio/tests.py [hunk 6]
```diff
@@ -343,13 +343,70 @@ def test_user_can_play_own_private_plio(self):
...
-        self.assertNotEqual(plio.id, response.data["id"])
-        self.assertNotEqual(plio.uuid, response.data["uuid"])
-        self.assertEqual(plio.name, response.data["name"])
+        for item in response.data.get("items", []):
+            new_item_ids.append(item["id"])
+            new_question_ids.append(item["details"]["id"])
+
+        response = self.client.get(
+            f"/api/v1/plios/{new_plio_uuid}/duplicate",
+        )
```
Comment (dalmia) [inline plio/tests.py:380]: why are you making a get request to duplicate?

File: plio/urls.py (hunk 1 above)
Comment (gurleen3) [reply plio/urls.py:65]: I thought that comment was not needed. Correcting it

File: plio/views.py [hunk 7]
```diff
@@ -261,11 +261,29 @@ def duplicate(self, request, uuid):
...
+
+        if items:
+            for index, _ in enumerate(items):
+                items[index].plio = plio
+                items[index].pk = None
+
+            # create the items
+            items = Item.objects.bulk_create(items)
+        # questions part
+        if questions:
```
Comment (dalmia) [inline plio/views.py:281]: questions with images haven't been handled as well?

File: plio/views.py (hunk 2 above)
Comment (gurleen3) [reply plio/views.py:751]: corrected

File: plio/views.py (hunk 3 above)
Comment (gurleen3) [reply plio/views.py:816]: corrected

File: plio/views.py (hunk 7 above)
Comment (gurleen3) [reply plio/views.py:294]: done in recent commit

**Pull request merged**